iceoryx_utils/include/iceoryx_utils/internal/cxx/forward_list.inl

```diff
@@ -0,0 +1,538 @@
+{
+    return m_size;
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::sizeType forward_list<T, CAPACITY>::capacity() const noexcept
+{
+    return CAPACITY;
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::sizeType forward_list<T, CAPACITY>::max_size() const noexcept
+{
+    return capacity();
+}
+
+
+template <typename T, int64_t CAPACITY>
+template <typename... ConstructorArgs>
+bool forward_list<T, CAPACITY>::emplace_front(ConstructorArgs&&... args) noexcept
+{
+    return (cend() != emplace_after(before_begin(), std::forward<ConstructorArgs>(args)...));
+}
+
+template <typename T, int64_t CAPACITY>
+template <typename... ConstructorArgs>
+typename forward_list<T, CAPACITY>::iterator
+forward_list<T, CAPACITY>::emplace_after(const_iterator afterToBeEmplacedIter, ConstructorArgs&&... args) noexcept
+{
+    if (this != afterToBeEmplacedIter.m_cList)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator of other list can't be used ");
+        std::terminate();
+    }
+
+    const sizeType l_toBeUsedIdx = getHeadFreeEl()->nextIdx;
+
+    if (m_size < CAPACITY && l_toBeUsedIdx < INVALID_INDEX)
+    {
+        // unlink from freeList
+        getHeadFreeEl()->nextIdx = getPointerToListNode()[l_toBeUsedIdx].nextIdx;
+
+        // data class c'tor
+        new (&(getPointerToListNode()[l_toBeUsedIdx].data)) T(std::forward<ConstructorArgs>(args)...);
+
+        // add to usedList
+        const sizeType emplaceAfterIdx = afterToBeEmplacedIter.m_iterListNodeIdx;
+        const sizeType emplaceBeforeIdx = getPointerToListNode()[emplaceAfterIdx].nextIdx;
+        getPointerToListNode()[l_toBeUsedIdx].nextIdx = emplaceBeforeIdx;
+        getPointerToListNode()[emplaceAfterIdx].nextIdx = l_toBeUsedIdx;
+
+        ++m_size;
+
+        return iterator{const_cast<forward_list<T, CAPACITY>*>(afterToBeEmplacedIter.m_cList), l_toBeUsedIdx};
+    }
+    else
+    {
+        errorMessage(__PRETTY_FUNCTION__, " capacity exhausted ");
+        return end();
+    }
+}
+
+
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::iterator
+forward_list<T, CAPACITY>::erase_after(const_iterator beforeToBeErasedIter) noexcept
+{
+    if (this != beforeToBeErasedIter.m_cList)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator of other list can't be used ");
+        std::terminate();
+    }
+
+    sizeType l_beforeToBeRmIdx = beforeToBeErasedIter.m_iterListNodeIdx;
+    if (l_beforeToBeRmIdx >= INVALID_INDEX)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator is end()");
+        return end();
+    }
+
+    sizeType l_toBeRmIdx = getPointerToListNode()[l_beforeToBeRmIdx].nextIdx;
+
+    if (l_toBeRmIdx < INVALID_INDEX && !empty())
+    {
+        // unlink from usedList
+        getPointerToListNode()[l_beforeToBeRmIdx].nextIdx = getPointerToListNode()[l_toBeRmIdx].nextIdx;
```

Comment:
Store `getPointerToListNode()[l_beforeToBeRmIdx]` in a variable explaining what it is.